engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/distribute/DeploymentDistributionBehavior.java

```diff
@@ -78,21 +80,37 @@ public void distributeDeploymentToPartition(
     final var deploymentPushedFuture =
         deploymentDistributor.pushDeploymentToPartition(key, partitionId, copiedDeploymentBuffer);
 
-    deploymentPushedFuture.onComplete(
```

Comment:
:heart_eyes: Thanks for getting rid of `runUntilDone`! I hope we can remove its usages completely soon!